service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -0,0 +1,388 @@
+package bio.terra.cbas.service;
+
+import static bio.terra.cbas.model.RunSetState.ERROR;
+import static bio.terra.cbas.model.RunSetState.RUNNING;
+import static bio.terra.cbas.models.CbasRunStatus.INITIALIZING;
+import static bio.terra.cbas.models.CbasRunStatus.QUEUED;
+import static bio.terra.cbas.models.CbasRunStatus.SYSTEM_ERROR;
+
+import bio.terra.cbas.common.DateUtils;
+import bio.terra.cbas.common.exceptions.DatabaseConnectivityException.RunCreationException;
+import bio.terra.cbas.common.exceptions.DatabaseConnectivityException.RunSetCreationException;
+import bio.terra.cbas.common.exceptions.InputProcessingException;
+import bio.terra.cbas.config.CbasApiConfiguration;
+import bio.terra.cbas.config.CbasContextConfiguration;
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.MethodVersionDao;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.dependencies.wds.WdsClientUtils;
+import bio.terra.cbas.dependencies.wds.WdsService;
+import bio.terra.cbas.dependencies.wds.WdsServiceApiException;
+import bio.terra.cbas.dependencies.wds.WdsServiceException;
+import bio.terra.cbas.dependencies.wes.CromwellService;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.model.RunSetState;
+import bio.terra.cbas.model.RunStateResponse;
+import bio.terra.cbas.models.CbasRunSetStatus;
+import bio.terra.cbas.models.CbasRunStatus;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.cbas.models.Run;
+import bio.terra.cbas.models.RunSet;
+import bio.terra.cbas.runsets.inputs.InputGenerator;
+import bio.terra.cbas.runsets.types.CoercionException;
+import bio.terra.cbas.util.UuidSource;
+import bio.terra.common.iam.BearerToken;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import cromwell.client.model.WorkflowIdAndStatus;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.databiosphere.workspacedata.model.RecordResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RunSetsService {
```

Comment:
this is the nit-iest pick I've ever picked, but I'm noticing that we sometimes name our files using the singular form `RunSet` and sometimes the plural `RunSets`. My vote is singular `RunSet`!

Reply:
Yeah I have seen that too but I tried to name the service this based on the controller and its called `RunSetsApiController`. I agree that we should just pick one, but I think that refactoring can be done in a later PR. If there is a strong opinion for singular I can rename `RunSetsService` to `RunSetService` but leave other name refactoring for another time.

Reply:
I'm honestly kind of embarrassed to point it out. It can definitely come later

Reply:
No its a completely valid point and I agree with the idea of being consistent in the naming